File: src/main/java/remake/internal/Bundle.java
package remake.internal;

import essentials.special.UTF8Control;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.text.MessageFormat;
import java.util.Locale;
import java.util.MissingResourceException;
import java.util.PropertyResourceBundle;
import java.util.ResourceBundle;

import static remake.Main.locale;
import static remake.Main.root;

public class Bundle {
    ResourceBundle RESOURCE_BUNDLE;

    public Bundle() {
        try {
            RESOURCE_BUNDLE = root.child("bundle.properties").exists()
                    ? new PropertyResourceBundle(Files.newInputStream(Paths.get(root.child("bundle.properties").path())))
                    : ResourceBundle.getBundle("bundle.bundle", locale, new UTF8Control());
        } catch (Exception e) {
            RESOURCE_BUNDLE = ResourceBundle.getBundle("bundle.bundle", new Locale("en", "US"), new UTF8Control());
        }
    }

    public Bundle(Locale locale) {
        try {
            RESOURCE_BUNDLE = root.child("bundle.properties").exists()
                    ? new PropertyResourceBundle(Files.newInputStream(Paths.get(root.child("bundle.properties").path())))
                    : ResourceBundle.getBundle("bundle.bundle", locale, new UTF8Control());
        } catch (Exception e) {
            RESOURCE_BUNDLE = ResourceBundle.getBundle("bundle.bundle", new Locale("en", "US"), new UTF8Control());
        }
    }

    public String get(String key, Object... params) {
        try {
            return MessageFormat.format(RESOURCE_BUNDLE.getString(key), params);
        } catch (MissingResourceException e) {
            return null;
        }
    }
}

File: src/main/kotlin/essentials/Bundle.kt
package essentials

import java.text.MessageFormat
import java.util.*

class Bundle {
    val resource : ResourceBundle

    constructor() {
        resource = ResourceBundle.getBundle("bundle", Locale.getDefault())
    }

    constructor(languageTag : String) {
        val locale = try {
            when(languageTag.substring(0, 2)) {
                "ko" -> Locale.KOREA
                else -> Locale.ENGLISH
            }
        } catch(e : Exception) {
            Locale.ENGLISH
        }
        resource = ResourceBundle.getBundle("bundle", locale)
    }

    operator fun get(key : String) : String {
        return MessageFormat.format(resource.getString(key))
    }

    operator fun get(key : String, vararg parameter : Any) : String {
        return MessageFormat.format(resource.getString(key), *parameter)
    }
}